root/usr/bin/discovery-menu

```diff
@@ -0,0 +1,576 @@
+#!/bin/ruby
+require "newt"
+require "discovery"
+require "facter"
+require "ipaddr"
+
+# disable GC until https://github.com/theforeman/ruby-newt/issues/2 is fixed
+GC.disable
+
+def fdi_version file = 'VERSION'
+  IO.read("/usr/share/fdi/#{file}").chomp rescue 'GIT'
+end
+
+def fdi_release file = 'RELEASE'
+  fdi_version file
+end
+
+def enable_root_account(pass)
+  command("echo 'root:#{pass}' | chpasswd && systemctl restart sshd.service")
+end
+
+def error_box(msg, debug_msg = nil)
+  log_err msg
+  ex_msg = debug_msg.to_s
+  if debug_msg.is_a? Exception
+    ex_bt = (debug_msg.to_s + "\n" + debug_msg.backtrace.join("\n")) rescue 'N/A'
+    log_err ex_bt
+  end
+  log_err ex_msg
+  Newt::Screen.centered_window(74, 20, "Fatal error")
+  f = Newt::Form.new
+  t_desc = Newt::Textbox.new(1, 1, 70, 13, Newt::FLAG_SCROLL)
+  random_pass = (1...10).map { (65 + rand(26)).chr }.join
+  t_desc.set_text "#{msg}:\n#{ex_msg}\n\n#{ex_bt}\n\n" +
+  "Once OK button is pressed, text user interface will be restarted\n" +
+  "and root account will be unlocked with the following random\n" +
+  "password:\n\n" +
+  "   #{random_pass}\n\n" +
+  "It is possible to use third console to login and investigate\n" +
+  "via journalctl and discovery-debug commands. "
+  b_ok = Newt::Button.new(66, 15, "OK")
+  f.add(b_ok, t_desc)
+  f.run
+  enable_root_account(random_pass)
+  exit(1)
+end
+
+def command(cmd)
+  log_msg "TUI executing: #{cmd}"
+  # do not run real commands in development (non-image) environment
+  return unless File.exist? '/usr/share/fdi/VERSION'
+  output = `#{cmd} 2>&1`
+  if $? != 0
+    error_box("Command failed: #{cmd}", output)
```

Comment:
Error box method is a fatal one, ends with `exit(1)`. This will lead to systemd restarting the service after some time.